tests/async_tests/test_interfaces.py

```diff
@@ -390,3 +390,36 @@ async def test_broken_socket_detection_many_open_files(
             assert status_code == 200
             assert ext == {"http_version": "HTTP/1.1", "reason": "OK"}
             assert len(http._connections[url[:3]]) == 1  # type: ignore
+
+
+@pytest.mark.anyio
+@pytest.mark.parametrize(
+    "url",
+    [
+        pytest.param((b"http", b"localhost", 12345, b"/"), id="connection-refused"),
+        pytest.param(
+            (b"http", b"doesnotexistatall.org", None, b"/"), id="dns-resolution-failed"
+        ),
+    ],
+)
+async def test_cannot_connect_tcp(backend: str, url) -> None:
```

Comment:
May be we can make it catchy, a-la `test_tcp_connection_errors_are_properly_wrapped`, like in BDD (then we can omit the comment)